### PR TITLE
Feature :: Add calendar

### PR DESCRIPTION
**Pull request by dmitriyVasilievich1986**
This pull request introduces a comprehensive calendar system for managing and calculating event-driven character stats across days. The main features include the new `Calendar` service, improved event handling (including support for special "empty" events), and enhancements to event filtering and stat calculation logic. Thorough tests and export organization are also provided.

**Calendar system and stat calculation:**

* Added a new `Calendar` class that manages an ordered sequence of `Day` instances, provides date-based lookup, builds event availability contexts, and recalculates stats across days with support for partial recalculation from a given date. (`src/services/calendar/Calendar.ts`, `src/services/calendar/index.ts`, `src/services/calendar/Calendar.test.ts`) [[1]](diffhunk://#diff-1663515d383c4f7b7d6ffe11778d0aa7a9cca06386779968f0330e9c50cbc295R1-R165) [[2]](diffhunk://#diff-41c9f57e366b2ac4789358786e2791c109d2bd425baeb241ec689963860ff981R1-R6) [[3]](diffhunk://#diff-f4db64d4fd20f725d2abe400610bc16f44a05bbb97f47f5036bf7d4d38a38fc7R1-R242)

**Event model and type system enhancements:**

* Introduced a new `EmptyEvent` special event, supporting its registration and type-safe usage throughout the event factory and event type system. (`src/services/event/models/specialEvents/EmptyEvent.ts`, `src/services/event/models/specialEvents/index.ts`, `src/services/event/models/specialEvents/types.ts`, `src/services/event/factory.ts`, `src/services/event/types.ts`) [[1]](diffhunk://#diff-f6d8d13d837f5915417d81d9d90823adde34de42849e5db10111bc1fbb3456c5R1-R18) [[2]](diffhunk://#diff-349b7051b0b0b5c53e1c1aaa17c61b5cc0406fb370597259cc7055a22e5e7f52R1-R8) [[3]](diffhunk://#diff-3773329b39da1ddecbcb8aa00d86f9486deb1a127ba2a45cab0054978b0dc886R1-R5) [[4]](diffhunk://#diff-d8df3add053d59f74dd3ca52c01b668c9723401d6344e52e6d6f11c117686294R8-R15) [[5]](diffhunk://#diff-d8df3add053d59f74dd3ca52c01b668c9723401d6344e52e6d6f11c117686294R24-R26) [[6]](diffhunk://#diff-f29ad38a0cf022c2411dbc72a00cc294ce410468171a780108fa858fc2b91313R3) [[7]](diffhunk://#diff-f29ad38a0cf022c2411dbc72a00cc294ce410468171a780108fa858fc2b91313R18)

**Event filtering and calculation improvements:**

* Enhanced `Day.filterEvents` to detect multiple events at the same time, optionally throwing an error or warning, and updated tests to cover these cases. (`src/services/day/Day.ts`, `src/services/day/Day.test.ts`) [[1]](diffhunk://#diff-0e95829a4da513230ceb432895f29076c1866ff664b21a5d1a9e5c4c005d5f92L130-R142) [[2]](diffhunk://#diff-fcb198fd16b18a04c635b66f3441ffbac4aae457f75039c7636932b3473a4ff0R238-R269)
* Improved stat calculation in `Day.calculateStats` to ensure correct event context is passed, supporting accurate chaining of stats across events and days. (`src/services/day/Day.ts`)

**Testing and exports:**

* Added comprehensive tests for the new calendar logic, including serialization, day lookup, event availability context, and stat calculation with various edge cases. (`src/services/calendar/Calendar.test.ts`)
* Organized and documented exports for the new calendar service. (`src/services/calendar/index.ts`)

### CODE REVIEW

File: src/services/calendar/Calendar.ts [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+        stats = day.statsAtEndOfDay;
+        payload.push(day);
+        continue;
+      }
+      const isAvailableProps = {
+        time: Times.Day, // not important
+        date: day.date,
+        stats,
+        event, // not important
+        currentDay: day,
```
[the remark this answers is not in the document]
Comment (dmitriyVasilievich1986) [reply src/services/calendar/Calendar.ts:130]: skip

File: src/services/event/models/specialEvents/types.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export const specialEventsNames = {
+  Empty: 'EmptyEvent',
+} as const;
+
+export type specialEventsNamesType = (typeof specialEventsNames)[keyof typeof specialEventsNames];
```
[the remark this answers is not in the document]
Comment (dmitriyVasilievich1986) [reply src/services/event/models/specialEvents/types.ts]: Renamed all enums to the PascalCase
